protos/dcs/mission/v0/mission.proto

```diff
@@ -434,12 +455,45 @@ message StreamUnitsResponse {
 
 message GetScenarioStartTimeRequest {}
 
-message GetScenarioStartTimeResponse {
-  string datetime = 1;
-}
+message GetScenarioStartTimeResponse { string datetime = 1; }
 
 message GetScenarioCurrentTimeRequest {}
 
-message GetScenarioCurrentTimeResponse {
-  string datetime = 1;
+message GetScenarioCurrentTimeResponse { string datetime = 1; }
+
+message AddMissionCommandRequest {
+  string id = 1;
+  string name = 2;
+  repeated string path = 3;
+
+  oneof target {
+    bool global = 4;
+    dcs.common.v0.Coalition coalition = 5;
+    string group = 6;
+  }
+}
+
+message AddMissionCommandResponse {}
+
+message RemoveMissionCommandItemRequest {
+  repeated string path = 1;
+  oneof target {
+    bool global = 2;
+    dcs.common.v0.Coalition coalition = 3;
+    string group = 4;
+  }
 }
+
+message RemoveMissionCommandItemResponse {}
+
+message AddMissionCommandSubMenuRequest {
+  string name = 1;
+  repeated string path = 2;
+  oneof target {
+    bool global = 3;
+    dcs.common.v0.Coalition coalition = 4;
+    string group = 5;
+  }
+}
+
+message AddMissionCommandSubMenuResponse {}
```

Comment:
Do you mean parity with the DCS API function signatures? This value can be trivially derived from the `Request` payload, so I don't think it makes sense to increase the surface area here.

Reply:
I would include it in the response. It is easy and if nothing else it allows the requester to optionally do a sanity check and protect against ED doing unexpected ED things like generating a different ID for _reasons_.

Reply:
Consider the following pseudo code snippets.
```
// cascading a path (the suggestion)
var lvl1 = grpcDcs.addSubMenu("First Level");
var lvl2 = grpcDcs.addSubMenu("Second Level", lvl1.path);
var menu = grpcDcs.addCommand("command", lvl2.path);
```
vs
```
// deriving the path
var path = [];
grpcDcs.addSubMenu("First Level", path);
path.push("First Level"); // or even extract the string as a local var...
const secondLevel = "Second Level";
grpcDcs.addSubMenu(secondLevel, [ "Or Maybe Prone to typo" ]);
path.push(secondLevel);
grpcDcs.addCommand("command", path);
```

So yes, I can see how it can be derived; it's not a show stopper. Both of us could agree if there is a minor change to our code, we make the client code feel neater when the effort is of similar nature.

https://github.com/DCS-gRPC/rust-server/blob/2db3b3bf4b1be0f7628a8171356198770d59dd89/src/rpc/atmosphere.rs#L8-L18
https://github.com/DCS-gRPC/rust-server/blob/15f806463cc7df23ac9c73d67c90fa0a8dbc1d76/lua/DCS-gRPC/methods/atmosphere.lua#L6-L10